kotlin-libraries-data/src/test/kotlin/com/baeldung/retrofit/EnumSerializationUnitTest.kt

```diff
@@ -0,0 +1,106 @@
+package com.baeldung.retrofit
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.junit.WireMockRule
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.converter.jackson.JacksonConverterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.POST
+import retrofit2.http.Path
+
+enum class OrderStatus { PENDING, COMPLETED, CANCELLED }
+
+data class Order(val id: String, val status: OrderStatus)
+
+interface OrderService {
+
+    @GET("orders/{id}")
+    suspend fun getOrder(@Path("id") orderId: String): Order
+
+    @POST("orders")
+    suspend fun createOrder(@Body order: Order): Response<Order>
+}
```

Comment:
Removing the `Response` wrapper to match

```suggestion
import org.junit.jupiter.api.Test
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.converter.jackson.JacksonConverterFactory
import retrofit2.converter.moshi.MoshiConverterFactory
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

enum class OrderStatus { PENDING, COMPLETED, CANCELLED }

data class Order(val id: String, val status: OrderStatus)

interface OrderService {

    @GET("orders/{id}")
    suspend fun getOrder(@Path("id") orderId: String): Order

    @POST("orders")
    suspend fun createOrder(@Body order: Order): Order
}
```